packages/flutter_tools/lib/src/build_system/build_system.dart

```diff
@@ -481,41 +480,59 @@ class _BuildInstance {
     final Stopwatch stopwatch = Stopwatch()..start();
     bool passed = true;
     bool skipped = false;
-    try {
-      final bool canSkip = await node.computeChanges(environment, fileCache);
+
+    // The build system should produce a list of aggregate input and output
+    // files for the overall build. The goal is to provide this to a hosting
+    // build system, such as Xcode, to configure logic for when to skip the
+    // rule/phase which contains the flutter build. When looking at the
+    // inputs and outputs for the individual rules, we need to be careful to
+    // remove inputs that were actually output from previous build steps.
+    // This indicates that the file is actual an output or intermediary. If
+    // these files are included as both inputs and outputs then it isn't
+    // possible to construct a DAG describing the build.
+    void updateGraph() {
+      for (File output in node.outputs) {
+        outputFiles[output.path] = output;
+      }
       for (File input in node.inputs) {
-        // The build system should produce a list of aggregate input and output
-        // files for the overall build. The goal is to provide this to a hosting
-        // build system, such as Xcode, to configure logic for when to skip the
-        // rule/phase which contains the flutter build. When looking at the
-        // inputs and outputs for the individual rules, we need to be careful to
-        // remove inputs that were actually output from previous build steps.
-        // This indicates that the file is actual an output or intermediary. If
-        // these files are included as both inputs and outputs then it isn't
-        // possible to construct a DAG describing the build.
         final String resolvedPath = input.resolveSymbolicLinksSync();
         if (outputFiles.containsKey(resolvedPath)) {
           continue;
         }
         inputFiles[resolvedPath] = input;
       }
+    }
+
+    try {
+      // If we're missing a depfile, wait until after evaluating the target to
+      // compute changes.
+      final bool canSkip = !node.missingDepfile &&
+        await node.computeChanges(environment, fileCache);
+
       if (canSkip) {
         skipped = true;
         printTrace('Skipping target: ${node.target.name}');
-        for (File output in node.outputs) {
-          outputFiles[output.path] = output;
-        }
+        updateGraph();
```

Comment:
Can you `return passed` here and unindent the `else`?

Reply:
Done